### PR TITLE
[kie-issues#628] Update repositories and other small changes to Apache

### PR DESCRIPTION
**Pull request by baldimir**
This PR removes JBoss Nexus from the repositories and replaces it with Apache Nexus. Then it fixes some other places with old references, like kiegroup etc.

### CODE REVIEW

**Comment by kie-ci3** [comment]
**PR job** `#1` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-drools/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-drools -u https://github.com/apache/incubator-kie-drools/pull/5553 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/drools/job/main/job/pullrequest_jobs/job/drools-pr/job/PR-5553/1/display/redirect

**Test results:**
- PASSED: 21722
- FAILED: 5

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/drools/job/main/job/pullrequest_jobs/job/drools-pr/job/PR-5553/1/testReport/org.drools.traits.compiler.integrationtests/TraitTypeGenerationTest/testNeeds/">org.drools.traits.compiler.integrationtests.TraitTypeGenerationTest.testNeeds</a></summary>
Exception executing consequence for rule "Main" in org.drools.abductive.test: java.lang.NullPointerException
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/drools/job/main/job/pullrequest_jobs/job/drools-pr/job/PR-5553/1/testReport/org.drools.traits.compiler.integrationtests/TraitTypeGenerationTest/testMvelJittingWithTraitProxies/">org.drools.traits.compiler.integrationtests.TraitTypeGenerationTest.testMvelJittingWithTraitProxies</a></summary>
Exception executing consequence for rule "In1" in org.drools.test: java.lang.NullPointerException
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/drools/job/main/job/pullrequest_jobs/job/drools-pr/job/PR-5553/1/testReport/org.drools.traits.compiler.integrationtests/TraitTypeGenerationTest/testIsAWith2KContainers/">org.drools.traits.compiler.integrationtests.TraitTypeGenerationTest.testIsAWith2KContainers</a></summary>
Exception executing consequence for rule "create student" in defaultpkg: java.lang.NullPointerException
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/drools/job/main/job/pullrequest_jobs/job/drools-pr/job/PR-5553/1/testReport/org.drools.traits.compiler.integrationtests/TraitTypeGenerationTest/testWithDeclaredTypeAndTraitInDifferentPackages/">org.drools.traits.compiler.integrationtests.TraitTypeGenerationTest.testWithDeclaredTypeAndTraitInDifferentPackages</a></summary>
Exception executing consequence for rule "R" in org.pkg3: java.lang.NullPointerException
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/drools/job/main/job/pullrequest_jobs/job/drools-pr/job/PR-5553/1/testReport/org.drools.traits.compiler.integrationtests/TraitTypeGenerationTest/testWithBeanAndTraitInDifferentPackages/">org.drools.traits.compiler.integrationtests.TraitTypeGenerationTest.testWithBeanAndTraitInDifferentPackages</a></summary>
Exception executing consequence for rule "R" in org.drools.test: java.lang.NullPointerException
</details>

File: bom/kie-efesto-bom/pom.xml [hunk 1]
```diff
@@ -21,13 +21,13 @@
...
-    <url>https://github.com/kiegroup/drools</url>
+    <connection>scm:git:https://github.com/apache/incubator-kie-drools.git</connection>
+    <developerConnection>scm:git:git@github.com:apache/incubator-kie-drools.git</developerConnection>
+    <url>https://github.com/apache/incubator-kie-drools</url>
   </scm>
   <issueManagement>
-    <system>jira</system>
-    <url>https://issues.jboss.org/browse/DROOLS</url>
+    <system>GitHub Issues</system>
+    <url>https://github.com/apache/incubator-kie-issues/issues</url>
```
Comment (jstastny-cz) [inline bom/kie-efesto-bom/pom.xml:30]: mailing lists section below lacks the newly added entry.

File: bom/kie-dmn-bom/pom.xml [hunk 2]
```diff
@@ -21,13 +21,13 @@
...
-    <url>https://github.com/kiegroup/drools</url>
+    <connection>scm:git:https://github.com/apache/incubator-kie-drools.git</connection>
+    <developerConnection>scm:git:git@github.com:apache/incubator-kie-drools.git</developerConnection>
+    <url>https://github.com/apache/incubator-kie-drools</url>
   </scm>
   <issueManagement>
-    <system>jira</system>
-    <url>https://issues.jboss.org/browse/DROOLS</url>
+    <system>GitHub Issues</system>
+    <url>https://github.com/apache/incubator-kie-issues/issues</url>
```
Comment (jstastny-cz) [inline bom/kie-dmn-bom/pom.xml:30]: mailing lists section below lacks the newly added entry.

**Review by jstastny-cz — APPROVED** [review]
2 minor comments about mailing list info, looks good.

File: bom/kie-dmn-bom/pom.xml (hunk 2 above)
Comment (baldimir) [reply bom/kie-dmn-bom/pom.xml:30]: Fixed.

File: bom/kie-efesto-bom/pom.xml (hunk 1 above)
Comment (baldimir) [reply bom/kie-efesto-bom/pom.xml:30]: Fixed.

**Review by mariofusco — APPROVED** [review]

**Pull request merged**